graylog2-server/src/main/java/org/graylog2/plugin/inputs/transports/util/KeyUtil.java

```diff
@@ -149,8 +149,14 @@ protected static PrivateKey loadPrivateKey(File file, String password) throws IO
                 throw new IllegalArgumentException("Unsupported key type");
             }
 
-            final KeyFactory kf = KeyFactory.getInstance("RSA");
-            return kf.generatePrivate(keySpec);
+            try {
+                final KeyFactory RSAkf = KeyFactory.getInstance("RSA");
+                return RSAkf.generatePrivate(keySpec);
+            } catch (InvalidKeySpecException e) {
+                LOG.debug("Generating RSA private key failed", e);
```

Comment:
IMO this should say that we are trying to generate a DSA key instead and not all hope is lost yet.